src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -155,32 +157,48 @@ void ScanAssembly(string assemblyPath, AssemblyScannerResults results, Dictionar
                 {
                     var skippedFile = new SkippedFile(assemblyPath, "Assembly does not reference at least one of the must referenced assemblies.");
                     results.SkippedFiles.Add(skippedFile);
-                    return;
+                    return false;
                 }
 
-                assembly = Assembly.LoadFrom(assemblyPath);
+                var loadedAssembly = Assembly.LoadFrom(assemblyPath);
 
-                if (results.Assemblies.Contains(assembly))
+                if (results.Assemblies.Contains(loadedAssembly))
                 {
-                    return;
+                    return false;
                 }
+
+                assembly = loadedAssembly;
+                return true;
             }
             catch (Exception ex) when (ex is BadImageFormatException || ex is FileLoadException)
             {
-                assembly = null;
                 results.ErrorsThrownDuringScanning = true;
 
                 if (ThrowExceptions)
                 {
                     var errorMessage = $"Could not load '{assemblyPath}'. Consider excluding that assembly from the scanning.";
                     throw new Exception(errorMessage, ex);
                 }
+                return false;
             }
+        }
 
+        void ScanAssembly(Assembly assembly, AssemblyScannerResults results)
+        {
             if (assembly == null)
             {
                 return;
             }
+
+            if (IsRuntimeAssembly(assembly.GetName()))
```

Comment:
I think we do not give any output for skipped runtime assemblies excluded by `IsRuntimeAssembly`? Also for IsIncluded, we just add them to the skipped list with a text, there is no log statement? The question of course is, do we need to add them to the skipped assemblies list?